rootfs/scheduler/__init__.py

```diff
@@ -667,7 +668,11 @@ def resolve_state(self, pod):
                 # is the pod ready to serve requests?
                 return states[container_status]
 
-        return states[pod['status']['phase']]
+        if pod['status']['phase'] in states:
+            return states[pod['status']['phase']]
+
+        # if no match was found for deis mapping then passthrough the real state
+        return pod['status']['phase']
```

Comment:
slight enhancement: we can make this all a two-liner with

```
pod_state = pod['status']['phase']
return states.get(pod_state, pod_state)
```

reducing the cyclomatic complexity by 1.